modules/ROOT/pages/application-deployment.adoc

```diff
@@ -0,0 +1,88 @@
+
+To enable parallel deployment, start Mule using the `-M-Dmule.deployment.parallel option`. For example:
+
+`mule -M-Dmule.deployment.parallel`
+
+All applications in the `/apps` directory start in parallel.
+
+== Undeploying Applications
+
+Don't delete the application folder directly, but rather delete an app's anchor file only.
+
+Deleting only the app's anchor file:
+
+* Prevents any interference from the hot-deployment layer and doesn't leave room for concurrent conflicting actions.
+* Avoids potential application jar locking issues on some operation systems and allows for clean shutdown and undeployment.
+
+For example, if the `stockTrader` app is running (app folder is there as well as the `$MULE_HOME/apps/stockTrader-anchor.txt` file), just delete the anchor file to remove the app from the Mule instance at runtime. Application folder is removed after the app terminates.
+
+[NOTE]
+After undeploying a Mule Application, there is a timeout of 15 seconds until LoggerContext stops. Log files for an application only release after this timeout expires. This information is important in Windows, where you can’t remove files that are in use by other processes.
+
+== Updating Applications at Runtime
+
+Updating a Mule application at runtime can be a complex change involving class modifications, endpoint modifications (for example, changing ports, etc), as well as reconfigured flows. As a result, any application update does a graceful app shutdown under the hood and reconfigures itself. In practice, this is pretty transparent to the user and happens within seconds.
+
+There are several ways you can update an application:
+
+* By dropping the modifications over an existing exploded app folder and touching the 'master' configuration file (`mule-config.xml` in app root by default).
+
+* By dropping a new `jar` version of the app into `$MULE_HOME/apps` dir. Mule detects this as an existing app update and ensures a clean redeployment of the app. Note that Mule discards any modifications to the old app folder - the new app folder is a clean exploded application from a jar.
+
+As you see, both integrate pretty well with existing build tools, the preference for one over another really depends on established development practices only.
+
+== Sharing Resources
+
+If you're deploying multiple applications to the same place and those applications could share the same resources, then you can create a common *domain* where you can define common configurations that can then be referenced by multiple projects. This allows you to, for example, expose different services in different projects through the same HTTP host and port and be able to deploy everything without any conflicts. See xref:shared-resources.adoc[Shared Resources].
```

Comment:
See xref:shared-resources.adoc[Shared Resources]. > This link can be sent to the bottom, in a "See Also" section.
You can also add a link to other mentioned topics like Hot Deployment in "See Also"